### ISSUE TITLE
Dashboard will show "My Interpretations" section and list any interpretations for curator

### ISSUE THREAD

**Issue by selinad** [ClinGen/clincoded#766]
The Dashboard will need a section called "My Interpretations" (much like My GDMs section).

Once a curator has an interpretation, the Dashboard will list all interpretations.  Possible examples:

- ClinVar name: no disease
- NC_: no disease
- ClinVar name: disease
- NC_: disease

**Comment by mrmin123** [comment]
@selinad @wrightmw 

Like this..?

![image](https://cloud.githubusercontent.com/assets/4326866/17716853/6de46a7e-63bf-11e6-9294-a2cc8dc64353.png)

**Comment by mrmin123** [comment]
![image](https://cloud.githubusercontent.com/assets/4326866/17717129/f978c9da-63c0-11e6-8fca-060f785cce73.png)
I don't know if you want the full title to display like this, or ellipsis it...

**Comment by selinad** [comment]
This looks great @mrmin123 !  I would vote for ellipsis....  thx!

**Comment by wrightmw** [comment]
Looks great @mrmin123 
I would go with ellipsis... it will make the interpretation history entries equal size and easier to scan..

**Comment by selinad** [comment]
HI @mrmin123 thx for checking.  I think we are OK for now - definitely not needed for this release.  We might want to see how it works and think about whether we want it in future (if that's ok?)

I noticed the "No disease associated" is on the same line as the variant in the screencapture above, but not earlier....did something change?

thx!
-S

**Comment by mrmin123** [comment]
@selinad Oops, fixed that. I also added ellipsis for long GDM titles..

![image](https://cloud.githubusercontent.com/assets/4326866/17749011/ddb91834-6470-11e6-811b-e29735c55955.png)

**Comment by selinad** [comment]
Looks super - thanks @mrmin123 !  

At some point, we will want to have a Status line - I think next release is a good time to think about what is "in progress" versus "complete" (or "final" or "preliminary interpretation")

**Comment by mrmin123** [comment]
@selinad @wrightmw instance here:
http://766-mc-vci-dashboard-4eebfcd-minchoi.instance.clinicalgenome.org/

This also has loading overlays on the dashboard, which we can hopefully use for #834 in the VCI

**Comment by mrmin123** [comment]
Also related to #722

**Comment by mrmin123** [comment]
@selinad @wrightmw Updated instance with both #766 and #905:
http://766-905-merge-9c88c90-minchoi.instance.clinicalgenome.org/

**Comment by selinad** [comment]
Hi @mrmin123 So sorry it took a bit to get to this - it's looking fantastic!  Your loading indicator seems to be working well also!

A few decision re functionality:
- it logs it as an interpretation if I start an interpretation but don't evaluate at all - we should consider whether this counts as an Interpretation - maybe it should require some evaluation?
- If possible, it would be nice to include a the record's "Status"
- in the history, it looks like the variant links to the record already in Interpretation mode rather than in Evidence View mode....I can see why you did that, and maybe it does make sense - what do others think?  My thought was it would go to Evidence View, but I'm not sure that's correct.  It may be better as you have it now.
- for the history that a variant is added, the variant is not linked to the Evidence view - could/should it be? (comments invited)

We probably just have a few decisions to make.

Does anyone have an opinion as to whether My Variant or My GDM records should be on top? (one over the other)?  I kinda like it as @mrmin123 has it....

Great stuff - thx @mrmin123 !

**Comment by mrmin123** [comment]
* We can certainly filter out the display of evaluation-less interpretations. What about in the Recent History view?
* Status can be displayed
* I thought it would make sense to link to the Interpretation mode since the history items (interpretation added, disease associated/changed, evaluation added/changed) relate to that interpretation, but we can definitely change this behavior
* I am not linking the variant because the variant can also be added via the GCI, and they share the same history display item. I didn't think it would make sense for a variant added in the GCI to link to the VCI, and since there's no way of differentiating, I left it out.

**Comment by mrmin123** [comment]
w/ Status:

![image](https://cloud.githubusercontent.com/assets/4326866/17951285/dbeec116-6a15-11e6-999f-5b1c3765041f.png)

**Comment by selinad** [comment]
Looks great, @mrmin123 !  Thanks for adding the Status.

Maybe @wrightmw has an opinion on the other items....

**Comment by wrightmw** [comment]
@selinad I like it the way @mrmin123 has it with Variants at the top, which is the same order they appear in the links to the resources on the LHS.

When I just viewed the evidence the entry showed as up in the list. I agree with @selinad that only viewing the evidence should not merit inclusion on this table. I would say that at least one evaluation has to be made before it shows up in the list.

**Comment by wrightmw** [comment]
@selinad @mrmin123 Also, I think linking to the Interpretation mode from the variant makes sense. If we agree to the point above (that at least one evaluation should be made for it to appear in the Interpretation history) then it makes sense are linked directly to the evaluations they have already made.

**Comment by selinad** [comment]
Thanks for weighing in, @wrightmw - this all sounds good!

**Comment by mrmin123** [comment]
@selinad @wrightmw the interpretation behavior was fixed here: http://766-905-merge-9c88c90-minchoi.instance.clinicalgenome.org/

Interpretation will not show up in history or My Interpretations list unless there's an evaluation associated with it.

The code represented in the instance (for #766 and #905) will not go in until after #925, however, since @jimmyzhen has a better implementation of some of the overlapping code, and I'd like to use that ultimately.

**Comment by selinad** [comment]
HI @mrmin123 works great!  Thank you for making that change.

One tiny, tiny thing: In Progress is in caps for gci and not for vci.  We should be consistent throughout site.  My preference is "In progress" but am open to others.

![screen shot 2016-08-29 at 5 17 33 pm](https://cloud.githubusercontent.com/assets/11320314/18071653/8a099fec-6e0c-11e6-843a-f3a961ff3fc4.png)

Also, alpha2 says "In Progress..." on the static bar, but this is now absent...
alpha2:
![screen shot 2016-08-29 at 5 20 50 pm](https://cloud.githubusercontent.com/assets/11320314/18071753/1660a102-6e0d-11e6-8fb4-1546659d8dd8.png)

instance:
![screen shot 2016-08-29 at 5 22 06 pm](https://cloud.githubusercontent.com/assets/11320314/18071764/243bfa56-6e0d-11e6-813b-2d52c0ace5b4.png)

**Comment by selinad** [comment]
I'm now a bit confused again - should it also be displaying curator name?
Also, the wrapping is funny - I will make a separate ticket

![screen shot 2016-08-29 at 5 39 25 pm](https://cloud.githubusercontent.com/assets/11320314/18072038/9e744d62-6e0f-11e6-82bd-ce9fce4a6e01.png)

**Comment by wrightmw** [comment]
I also see odd wrapping:
![screen shot 2016-08-29 at 5 51 13 pm](https://cloud.githubusercontent.com/assets/15131169/18072223/4c787ae0-6e11-11e6-92f1-c41d12158a75.png)

and if I refresh the page, I get this:
![screen shot 2016-08-29 at 5 50 51 pm](https://cloud.githubusercontent.com/assets/15131169/18072232/5abb78b4-6e11-11e6-9a7e-096c7ac7ead1.png)

**Comment by wrightmw** [comment]
@mrmin123 The functionality is great. Thanks!

**Comment by mrmin123** [comment]
@selinad Once you save an evaluation in an interpretation, its status should go from 'In Progress' to 'Evaluation'. The casing of the status text on the dashboard has been fixed.

The user's name has never been shown under 'My Interpretations'.

@selinad @wrightmw I can't re-create the odd wrapping, even on the same instance... Could one of you show it to me on your machines when you get the chance?

Also I have no idea how you got to the 'Cannot read property vertebrate of undefined'... the above two might issues might actually have something to do with the instance, more than anything...

**Comment by mrmin123** [comment]
@wrightmw The probable cause of the refresh bug was found and fixed by @jimmyzhen in #959

**Comment by mrmin123** [comment]
@selinad @wrightmw there's a new instance here: http://766-905-merge-4740902-minchoi.instance.clinicalgenome.org/dashboard/ that has the new code merged in.

You'll notice that the spinner areas on the dashboard are now white, before changing to gray... this is because I'm re-using the code used in the VCI area. Let me know if this is acceptable or if you want it changed.

**Comment by wrightmw** [comment]
@mrmin123 The spinner looks great to me

**Comment by mrmin123** [comment]
@selinad any thoughts?

**Comment by selinad** [comment]
Hi @mrmin123 My apologies for the delay.  

* Spinner works well - loaded pretty quickly, but caught a glimpse of it!
* History is being recorded correctly - when no evaluation saved, regardless of whether one is "in" an Interpretation, it does not create an Interpretation under "My Interpretations" or as part of history
* When evaluation is saved, correct functionality under My Interpretations and in history.

Nice!!

Question:
To match the GDM display, could the disease be associated with the variant in one line? If so, should I include on the ticket I'm planning to add about syncing history behavior/display throughout vci?
![image](https://cloud.githubusercontent.com/assets/11320314/18219805/7c058a42-7121-11e6-97e7-9774a851227e.png)

**Comment by kilodalton** [comment]
[pls ignore: comment added only to refresh PR display on waffle]

**Comment by selinad** [comment]
@mrmin123  - this looks good except the status says "Evaluation" instead of In progress...shouldn't it say "In progress?"  I know there was some confusion around this (to which I contributed) and we changed a few things, but we only use "In progress" and "Provisional" now.....

**Comment by mrmin123** [comment]
@selinad A ticket for that already exists as #974. It required further investigation and conversation as to how the `status` field is set in the database and will be addressed in the next release.

**Comment by selinad** [comment]
ok, thx for the reminder about that ticket and the update.

**Comment by wrightmw** [comment]
Looks good, except for issue to be resolved in #974 in next release.

**Comment by kilodalton** [comment]
Included in the last release (R7). This included the Variant Curation Interface. Nice job and thanks for your hard work.
